### PR TITLE
feat: add matcher `toHaveStatus`

### PR DESCRIPTION
**Pull request by Tamir-M**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -294,6 +295,23 @@ test('passes when using .not.toHave5xxStatus() for 200', async () => {
 });
 ```
 
+#### .toHaveStatus()
+
+Use `.toHaveStatus` when checking if response has a specific status
+
+```js
+test('passes when response have status code 500', async () => {
```
Comment (rluvaton) [inline README.md:303]: this is not really correct

Comment (Tamir-M) [reply README.md]: Fixed

File: test/matchers/status/generic/toHaveStatus.test.js [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+              } catch (e) {
+                t.assert.snapshot(e);
+              }
+            });
+          }
+        });
+      });
+
+      describe('.not.toHaveStatus', () => {
+        test('passes when the actual status is different to the status code', async () => {
```
Comment (rluvaton) [inline test/matchers/status/generic/toHaveStatus.test.js:58]:
```suggestion
        test('passes when the actual status is different than the expected status code', async () => {
```

File: test/matchers/status/generic/toHaveStatus.test.js [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+              {},
+            );
+
+            expect(response).not.toHaveStatus(status - 1);
+          }
+        });
+
+        describe('status 200 to 599', function allTests() {
+          for (let status = 200; status <= 599; status++) {
+            test(`fails when response have status code ${status} and the actual response type is the same`, async (t) => {
```
Comment (rluvaton) [inline test/matchers/status/generic/toHaveStatus.test.js:74]: it's not really the response **type**

File: types/shared.d.ts [hunk 4]
```diff
@@ -51,4 +51,10 @@ export interface SharedMatchers<R> {
    * Use .toHave5xxStatus when checking if Axios response status code is between 500 and 599 included
    */
   toHave5xxStatus(): R;
+
+  /**
+   * Use .toHave5xxStatus when checking if HTTP response status code is between 500 and 599 included
```
[the remark this answers is not in the document]
Comment (Tamir-M) [reply types/shared.d.ts:56]: My bad, done.

File: test/matchers/status/generic/toHaveStatus.test.js (hunk 3 above)
Comment (Tamir-M) [reply test/matchers/status/generic/toHaveStatus.test.js:74]: You are correct.

**Review by rluvaton — APPROVED** [review]

**Pull request merged**